clang/lib/Basic/Targets/AVR.h

```diff
@@ -57,7 +57,8 @@ class LLVM_LIBRARY_VISIBILITY AVRTargetInfo : public TargetInfo {
     Int16Type = SignedInt;
     Char32Type = UnsignedLong;
     SigAtomicType = SignedChar;
-    resetDataLayout("e-P1-p:16:8-i8:8-i16:8-i32:8-i64:8-f32:8-f64:8-n8-a:8");
+    resetDataLayout(
+        "e-P1-p:16:8-i8:8-i16:8-i32:8-i64:8-f32:8-f64:8-n8-n16:8-a:8");
```

Comment:
If this change has partial improvement on https://github.com/llvm/llvm-project/issues/151080, you can submit this without the TTI part.

Sub optimal results are also accepted.

Reply:
So a new test to show `Loop Terminator Folding (loop-term-fold)` is also needed to show the necessity of adding TTI.